src/libostree/ostree-sysroot-deploy.c

```diff
@@ -2218,24 +2217,23 @@ swap_bootloader (OstreeSysroot *sysroot, OstreeBootloader *bootloader, int curre
   if (!_ostree_sysroot_ensure_boot_fd (sysroot, error))
     return FALSE;
 
+  g_assert_cmpint (sysroot->boot_fd, !=, -1);
+
   /* The symlink was already written, and we used syncfs() to ensure
    * its data is in place.  Renaming now should give us atomic semantics;
    * see https://bugzilla.gnome.org/show_bug.cgi?id=755595
    */
   if (!glnx_renameat (sysroot->boot_fd, "loader.tmp", sysroot->boot_fd, "loader", error))
     return FALSE;
 
-  /* Now we explicitly fsync this directory, even though it
-   * isn't required for atomicity, for two reasons:
-   *  - It should be very cheap as we're just syncing whatever
-   *    data was written since the last sync which was hopefully
-   *    less than a second ago.
-   *  - It should be sync'd before shutdown as that could crash
-   *    for whatever reason, and we wouldn't want to confuse the
-   *    admin by going back to the previous session.
-   */
-  if (fsync (sysroot->boot_fd) != 0)
-    return glnx_throw_errno_prefix (error, "fsync(boot)");
+  /* As grub doesn't support replaying XFS journal, we must fsfreeze/thaw again here */
```

Comment:
I'd like to not lose the previous comment here because it was really describing the intention of the code.

Reply:
Something like
```
  /* As grub doesn't support replaying XFS journal,
   * we must fsfreeze/thaw again here:
   *  - It should be very cheap as we're just syncing whatever
   *    data was written since the last sync which was hopefully
   *    less than a second ago.
   *  - It should be sync'd before shutdown as that could crash
   *    for whatever reason, and we wouldn't want to confuse the
   *    admin by going back to the previous session.
   */
```
?

Reply:
done